.github/workflows/build_and_test.yml

```diff
@@ -3533,6 +3540,8 @@ jobs:
     steps:
     - name: install xerces
       run: sudo apt-get -y install libxerces-c-dev
+    - name: update apt
+      run: sudo apt-get update
```

Comment:
Shouldn't this go before installing xerces?

Reply:
I originally had it that way (see commit history), but it would be a change from how we were doing it before and, more importantly, it would mean we might potentially get different versions of xerces between what was used by the cached ACE artifact and what's used by the test & build jobs. If we know they'll be ABI compatible, it might not matter, but I was worried it might expect the same version.

On the other hand, I don't know that not upgrading apt would even avoid this issue 100%, since the virtual environment runner can still get upgraded out from underneath us (and so the "default" version of xerces may change). But this approach seemed like it might be the safest bet to sidestep the issue for cached ACE artifacts & xerces versions.  Happy to try it the other way if we don't think there will be any issues...